File: algorithms/1790.check-if-one-string-swap-can-make-strings-equal/solution.rb
# @param {String} s1
# @param {String} s2
# @return {Boolean}
def are_almost_equal(s1, s2)
  a, b = [], []
  (0..s1.size).each do |i|
    unless s1[i] == s2[i]
      a << s1[i]
      b << s2[i]
      return false if a.size > 2
    end
  end
  a.sort == b.sort
end

File: algorithms/1790.check-if-one-string-swap-can-make-strings-equal/solution.go
func areAlmostEqual(s1 string, s2 string) bool {
	a, b := []rune{}, []rune{}
	runes1, runes2 := []rune(s1), []rune(s2)

	for i := 0; i < len(s1); i++ {
		if s1[i] != s2[i] {
			a = append(a, runes1[i])
			b = append(b, runes2[i])

			if len(a) > 2 {
				return false
			}
		}
	}

	sort.Slice(a, func(i, j int) bool {
		return a[i] < a[j]
	})
	sort.Slice(b, func(i, j int) bool {
		return b[i] < b[j]
	})
	return string(a) == string(b)
}
